x/ccv/provider/keeper/keeper.go

```diff
@@ -621,6 +621,70 @@ func (k Keeper) DeleteInitChainHeight(ctx sdk.Context, chainID string) {
 	store.Delete(types.InitChainHeightKey(chainID))
 }
 
+// SetPendingVSCs sets in store the list of pending ValidatorSetChange packets under chain ID
+func (k Keeper) SetPendingVSCs(ctx sdk.Context, chainID string, packets []ccv.ValidatorSetChangePacketData) {
+	store := ctx.KVStore(k.storeKey)
+	var data [][]byte
+	for _, p := range packets {
+		pdata, err := p.Marshal()
+		if err != nil {
+			panic("failed to marshal ValidatorSetChange packet data")
+		}
+		data = append(data, pdata)
+	}
+	buf := &bytes.Buffer{}
+	err := json.NewEncoder(buf).Encode(data)
+	if err != nil {
+		panic("failed to encode json")
+	}
+	store.Set(types.PendingVSCsKey(chainID), buf.Bytes())
+}
+
+// GetPendingVSCs returns the list of pending ValidatorSetChange packets stored under chain ID
+func (k Keeper) GetPendingVSCs(ctx sdk.Context, chainID string) (packets []ccv.ValidatorSetChangePacketData, found bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.PendingVSCsKey(chainID))
+	if bz == nil {
+		return nil, false
+	}
+	buf := bytes.NewBuffer(bz)
+
+	var data [][]byte
+	json.NewDecoder(buf).Decode(&data)
+
+	for _, pdata := range data {
+		var p ccv.ValidatorSetChangePacketData
+		err := p.Unmarshal(pdata)
+		if err != nil {
+			panic("failed to unmarshal ValidatorSetChange packet data")
+		}
+		packets = append(packets, p)
+	}
+
+	return packets, true
+}
+
+// AppendPendingVSC adds the given ValidatorSetChange packet to the list
+// of pending ValidatorSetChange packets stored under chain ID
+func (k Keeper) AppendPendingVSC(ctx sdk.Context, chainID string, packet ccv.ValidatorSetChangePacketData) {
```

Comment:
nit: this function seems like it can be merged with `SetPendingVSC`, if you take in a list and always append to existing list if it exists